hw/top_earlgrey/data/ip/chip_otp_ctrl_testplan.hjson

```diff
@@ -0,0 +1,208 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  name: chip_otp_ctrl
+  testpoints: [
```

Comment:
There are many missing tests, like
- DAI readability: locked by CSR of digest
- DAI writeability, by digest
- Digest update, either via DAI or explicitly by SW
- Integrity checks, period and timeout, and explicit via CHECK_TRIGGER and CHECK_TRIGGER_REGWEN
- SW_CFG_WINDOW

Reply:
I think it is worth creating a digest update testpoint.
This is still missing testpoints for the integrity checks.

Reply:
Can you elaborate? you mean verify digest?

Reply:
I mean the digest generation and check that it prevents accesses: hw_cfg and secret partitions generate the digest in hardware as described in [https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#initialization](https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#digest-calculation-sequence), other partitions generate the digest in SW. I assume the manufacturing and personalization tests will do this, and digest update is a one-time operation, so the only additional test may be to check once a digest is generated the partition is not writeable, and only secret partitions are not readable based on the digest.

The [integrity and consistency checks](https://opentitan.org/book/hw/ip/otp_ctrl/doc/theory_of_operation.html#partition-checks) are programmed via CSRs as described in [https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#initialization](https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#initialization)